apps/web/src/utils/templates.ts

```diff
@@ -0,0 +1,13 @@
+import { IconName } from '@fortawesome/fontawesome-svg-core';
+
+export const getWorkflowBlueprintDetails = (templateName: string): { name: string; iconName: IconName } => {
+  const regexResult = /^:.{1,}:/.exec(templateName);
+  let name = '';
+  let iconName = 'fa-solid fa-question';
+  if (regexResult !== null) {
+    name = templateName.replace(regexResult[0], '').trim();
+    iconName = regexResult[0].replace(/:/g, '').trim();
+  }
+
+  return { name, iconName: iconName as IconName };
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** I realize this is just refactoring / moving this, but this IconName coupling is going to be a problem with de-coupling from our icon set(s)‏

Reply:
what do you mean by `our icon set(s)`?

small note: this logic was created so we could store icons by string conversion, this way we can store them easily in the template store under workflow.name

Reply:
Yeah, this is a tough one, but we're tightly coupled with FontAwesome as an icon set due to this implementation